### PR TITLE
Fix code signing in Xcode 8

### LINKED ISSUE #5523: Codesigning in Xcode 8
* [ x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Run `pod install` or `pod update`

## What happened instead?

Pods are installed but there is no Development Team set. You must select every target in the Pods project and set the team

### Stack

```
   CocoaPods : 1.0.1
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.12 (16A201w)
       Xcode : 7.3.1 (7D1014)
         Git : git version 2.7.4 (Apple Git-66)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ fdeab0150c4f811dc33db08c74e66cb75b54697e
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.0.0
```

### Podfile

```ruby
# Uncomment this line to define a global platform for your project
platform :ios, '9.0'

use_frameworks!
Target 'some appt' do
pod 'Onboard'
pod 'Localytics'
pod 'BuddyBuildSDK'
pod 'Doorbell'
pod 'Google/Analytics'
pod "LaunchKit"
pod "Branch"
pod 'Charts', :git => "https://github.com/opswhisperer/Charts.git", :branch => "xcode8_swift23"
pod "AFNetworking"
pod 'FBSDKCoreKit'
pod 'FBSDKShareKit'
pod 'FBSDKLoginKit'
pod 'iRate'
pod 'GoogleConversionTracking'
pod 'UIBarButtonItem-Badge', :git => "https://github.com/mikeMTOL/UIBarButtonItem-Badge.git"
pod 'GoogleIDFASupport'
end
```

## Project that demonstrates the issue

any project

### PR DESCRIPTION
**Pull request by neonichu**
This should fix building Pods for device using Xcode 8 beta.

Fixes #5523

### CODE REVIEW

**Comment by DanToml** [comment]
LGTM - Glad that it somewhat keeps old behaviour when `ProvisioningStyle` is missing.

**Comment by DanToml** [comment]
When you rebase this, it'll need to go in `Installer::Xcode::PodsProjectGenerator#prepare`

File: lib/cocoapods/installer.rb [hunk 1]
```diff
@@ -594,6 +594,8 @@ def prepare_pods_project
           build_configuration.build_settings['TVOS_DEPLOYMENT_TARGET'] = tvos_deployment_target.to_s if tvos_deployment_target
           build_configuration.build_settings['STRIP_INSTALLED_PRODUCT'] = 'NO'
           build_configuration.build_settings['CLANG_ENABLE_OBJC_ARC'] = 'YES'
+          build_configuration.build_settings['CODE_SIGNING_REQUIRED'] = 'NO'
+          build_configuration.build_settings['PROVISIONING_PROFILE_SPECIFIER'] = 'AAAAAAAAAA/'
```
Comment (orta) [inline lib/cocoapods/installer.rb]: Open up and say `AAAAAAA\`

Giphy

Comment (mrackwitz) [inline lib/cocoapods/installer.rb]: I think we should use a self-explanatory value here, which can referred to (e.g. `NO_SIGNING/` if underscores would be acceptable).

Comment (neonichu) [inline lib/cocoapods/installer.rb]: or `YAATUSABES`

**Comment by segiddins** [comment]
👍 with a rebase, test coverage, and a changelog

**Comment by mrackwitz** [comment]
:+1: beside my comment…

**Comment by opswhisperer** [comment]
This is not working for me

screenshot

**Comment by neonichu** [comment]
@opswhisperer looks as if you're only using this branch, you also need CocoaPods/Xcodeproj#387 — also the error in Xcode's General tab will not go away — we can only fix building at the moment.

**Comment by opswhisperer** [comment]
Ok, missed that. Even with that I'm getting a different error now than before
`No signing certificate 'iOS Development' found:  No 'iOS Development' signing certificate with a private key was found.`
For each one of my targets

**Comment by opswhisperer** [comment]
Never mind all good!

**Comment by warpling** [comment]
Is Travis stuck?

**Comment by neonichu** [comment]
I'm not quite sure about the result differences in the validator unit specs and also the integration specs:

```diff
-          LD_RUNPATH_SEARCH_PATHS: $(inherited) @executable_path/Frameworks @loader_path/Frameworks
+          LD_RUNPATH_SEARCH_PATHS:
+            - $(inherited)
+            - '@executable_path/Frameworks'
+            - '@loader_path/Frameworks'
```

```diff
-          ARCHS: $(ARCHS_STANDARD_64_BIT)
-          CODE_SIGN_IDENTITY[sdk=iphoneos*]: iPhone Developer
+          ARCHS:
+            - $(ARCHS_STANDARD_64_BIT)
+          CODE_SIGN_IDENTITY[sdk=iphoneos*]: ''
```

Seems like some array-y things become single strings and vice versa when using Xcode 8 — would appreciate a second set of eyes to determine if this is something to be addressed in Xcodeproj or not.

File: lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb [hunk 2]
```diff
@@ -45,12 +45,13 @@ def target_definition
...
-              'PODS_ROOT'                 => '$(SRCROOT)',
-              'PRODUCT_BUNDLE_IDENTIFIER' => 'org.cocoapods.${PRODUCT_NAME:rfc1034identifier}',
-              'SKIP_INSTALL'              => 'YES',
+              'MACH_O_TYPE'                       => 'staticlib',
+              'OTHER_LDFLAGS'                     => '',
+              'OTHER_LIBTOOLFLAGS'                => '',
+              'PODS_ROOT'                         => '$(SRCROOT)',
+              'PRODUCT_BUNDLE_IDENTIFIER'         => 'org.cocoapods.${PRODUCT_NAME:rfc1034identifier}',
+              'SKIP_INSTALL'                      => 'YES',
+              'CODE_SIGN_IDENTITY[sdk=iphoneos*]' => '',
```
Comment (segiddins) [inline lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb]: please keep the list sorted

**Comment by segiddins** [comment]
@neonichu if it still works in Xcode 7 as an array, then I wouldn't worry about it

File: lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb (hunk 2 above)
Comment (neonichu) [inline lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb]: Ah right, didn't notice that the list was sorted :)

**Comment by mrackwitz** [comment]
>@neonichu if it still works in Xcode 7 as an array, then I wouldn't worry about it

If the single values are managed together as strings as seen from the changes in the unit tests, this could potentially lead to issues when relying on e.g. `uniq` to avoid duplicate values. What has caused this change?

**Comment by neonichu** [comment]
@mrackwitz Using Xcode 8 instead of 7.3.1 to run the tests — no changes to CP except the ones in this PR when running them.

**Comment by neonichu** [comment]
I reverted the unit test changes, since we're still running those against 7.3 on CI and updated the integration specs in CocoaPods/cocoapods-integration-specs#74

This should be good to go now 👍

**Comment by neonichu** [comment]
Travis 👷 finished now 👌

**Comment by tsheaff** [comment]
When i this getting merged to master?

**Comment by opswhisperer** [comment]
We also need CocoaPods/Xcodeproj#387 which seems to have been deleted.

**Comment by neonichu** [comment]
@opswhisperer that shouldn't be needed anymore, the settings where transferred to CP itself in order to not mess with usage of `xcodeproj` outside of the CP context.

**Comment by opswhisperer** [comment]
Odd, because I have 1.0.1 and I'm still having code signing errors. It looks as if the settings are correct but I get build errors. I forced a reinstall of xcodeproj because I had the deleted PR installed locally.

```
Signing for AFNetworking requires a development team. Select a development team in the Target Editor.
Code signing is required for product type 'Framework' in SDK 'iOS 10.0'
```

**Comment by neonichu** [comment]
This isn't included in 1.0.1 — it has just been merged to master so far

**Comment by opswhisperer** [comment]
Ahh that makes sense, thanks

**Comment by aurawindsurfing** [comment]
Hi, can you point to me where I find info how to install those not relesed changes? Should I simply pull this repo? How will that work with existing install? I'm sorry for asking, just had trouble finding the info on the obvious. Cheers!

**Comment by orta** [comment]
There is a guide on this: https://guides.cocoapods.org/using/a-gemfile.html

**Comment by tal-sh** [comment]
My project contains multiple targets, and different signing identities. Is something like that could be supported by xcode?

**Comment by pbernery** [comment]
Don't know if I'll get an answer now :) : why did you set the `PROVISIONING_PROFILE_SPECIFIER` to `NO_SIGNING/`? 
Is this expected to be changed by the developer with its own prov. profile?

**Comment by DanToml** [comment]
@pbernery We sign the frameworks in the embed frameworks script

**Comment by pbernery** [comment]
Thanks @DanToml for this quick response.

I am trying to find the right solution to a signing issue related to this. I have the following error when building on the command line (used on my CI):
```
=== BUILD TARGET Alamofire OF PROJECT Pods WITH CONFIGURATION Release ===

Check dependencies
Alamofire does not support provisioning profiles. Alamofire does not support provisioning profiles, but provisioning profile NO_SIGNING/ has been manually specified. Set the provisioning profile value to "Automatic" in the build settings editor.
```
I got this error for all Pods. Removing "NO_SIGNING/" fixes this issue and lets Xcode continue.

I did not find any issue in the CocoaPods GitHub related to "NO_SIGNING/", and am trying to understand the build/sign workflow expected by CP. 

The dependency resolution of Xcode makes the Pods project targets compile first. It seems that Xcode 8.3.2 checks provisioning profiles as soon as something is set in this field, even though "Don't code sign" is specified in these projects. Is this a changed behaviour since the inclusion of this patch or is there something wrong in my project?

I made a `pod deintegrate && pod install` to make sure I did not changed the Pods project mistakenly.

**Comment by Daniate** [comment]
@pbernery I got this error too.

**Pull request merged**